src/main/java/com/epam/aidial/core/service/ResourceOperationService.java

```diff
@@ -28,25 +26,16 @@ public void moveResource(String bucket, String location, ResourceDescription sou
             throw new IllegalArgumentException("Moving folders is not supported");
         }
 
-        String sourceResourcePath = source.getAbsoluteFilePath();
         String sourceResourceUrl = source.getUrl();
-        String destinationResourcePath = destination.getAbsoluteFilePath();
         String destinationResourceUrl = destination.getUrl();
 
-        if (!hasResource(source)) {
+        if (!resourceService.hasResource(source)) {
             throw new IllegalArgumentException("Source resource %s do not exists".formatted(sourceResourceUrl));
         }
 
         ResourceType resourceType = source.getType();
         switch (resourceType) {
-            case FILE -> {
-                if (!overwriteIfExists && storage.exists(destinationResourcePath)) {
-                    throw new IllegalArgumentException("Can't move resource %s to %s, because destination resource already exists"
-                            .formatted(sourceResourceUrl, destinationResourceUrl));
-                }
-                storage.copy(sourceResourcePath, destinationResourcePath);
-            }
-            case CONVERSATION, PROMPT, APPLICATION -> {
+            case FILE, CONVERSATION, PROMPT, APPLICATION -> {
```

Comment:
simplify by validating resourceType in a separate method

Reply:
Done.